src/core/Akka/Actor/FSM.cs

```diff
@@ -1336,7 +1337,7 @@ protected virtual void LogTermination(Reason reason)
                 {
                     _log.Error(failure.Cause.AsInstanceOf<Exception>(), "terminating due to Failure");
                 }
-                else
+                else if(!(failure.Cause is null))
```

Comment:
This is a bad way of doing this. A failure is a failure, you should not block the error log.
A better way to handle this is to use the string `"null"` if `failure.Cause` is `null`.

Reply:
Done in the latest commit